src/main/java/com/aws/greengrass/lifecyclemanager/Lifecycle.java

```diff
@@ -800,9 +800,11 @@ void setClosed(boolean b) {
      * Start Service.
      */
     final void requestStart() {
-        // Ignore start requests if the service is closed
-        if (isClosed.get()) {
-            return;
+        // If the lifecycle thread is still running when isClosed is true, then it must be still executing
+        // the shutdown lifecycle. In this case it's ok to start service again.
+        if (isClosed.compareAndSet(true, false)) {
+            logger.atWarn("service-shutdown-interrupted")
```

Comment:
does shutdown really get interrupted or does it finish and then the service is started up again?

Reply:
shutdown lifecycle completes and state transitions to INSTALLED; other cleanups are skipped - such as removing service topics or recycling lifecycle thread

Reply:
But what's skipped is expected in this case, this doesn't sound like anything gets interrupted to me, you could maybe say that the service was originally closing, but now it won't due to a subsequent restart request